src/cli/lambdapi.ml

```diff
@@ -79,11 +79,15 @@ let decision_tree_cmd : Config.t -> (Path.t * string) -> unit =
     let sym =
       let sign = Compile.compile false mp in
       let ss = Sig_state.of_sign sign in
-      let (prt, prv) = (true, true) in
-      try Sig_state.find_sym false ss (Pos.none ([], sym)) ~prt ~prv
+      let ss = Sig_state.open_sign ss sign in
+      try Sig_state.find_sym false ss (Pos.none ([], sym)) ~prt:true ~prv:true
       with Not_found -> fatal_no_pos "Symbol not found."
     in
-    out 0 "%a" Tree_graphviz.to_dot sym
+    if Timed.(!(sym.sym_rules)) = [] then
+      wrn None "Cannot print decision tree: \
```

Comment:
Why not simply say: "no dot file generated since symbol ... has no rule" (because we could output an empty dot file instead)?

Reply:
Do you mean without warning? The command does not generate a file, it prints the graph on the standard output.

Reply:
No, I am just talking about the message.

Reply:
It's meaningless to say that no file has been generated, since it never generates files...

Reply:
Ah it's printed on stdout now.